addons/nuxeo-retention/nuxeo-retention-core/src/main/java/org/nuxeo/retention/listeners/RetentionDocumentEventListener.java

```diff
@@ -0,0 +1,125 @@
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.EventBundle;
+import org.nuxeo.ecm.core.event.EventContext;
+import org.nuxeo.ecm.core.event.PostCommitFilteringEventListener;
+import org.nuxeo.ecm.core.event.impl.DocumentEventContext;
+import org.nuxeo.retention.RetentionConstants;
+import org.nuxeo.retention.adapters.Record;
+import org.nuxeo.retention.adapters.RetentionRule;
+import org.nuxeo.retention.service.RetentionManager;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Listener processing events with a {@link org.nuxeo.ecm.core.event.impl.DocumentEventContext}). If the targeted
+ * document is a record and has an attached event-based, the rule is then evaluated to check if the retention expiration
+ * date has to be recomputed.
+ *
+ * @since 11.1
+ */
+public class RetentionDocumentEventListener implements PostCommitFilteringEventListener {
+
+    private static final Logger log = LogManager.getLogger(RetentionDocumentEventListener.class);
+
+    @Override
+    public boolean acceptEvent(Event event) {
+        EventContext eventCtx = event.getContext();
+        if (!(eventCtx instanceof DocumentEventContext)) {
+            return false;
+        }
+        RetentionManager retentionManager = Framework.getService(RetentionManager.class);
+        if (retentionManager == null) {
+            log.trace("RetentionManager not started yet?!");
+            return false;
+        }
+        return retentionManager.getAcceptedEvents().contains(event.getName());
+    }
+
+    @Override
+    public void handleEvent(EventBundle events) {
+
+        RetentionManager retentionManager = Framework.getService(RetentionManager.class);
+
+        Map<String, Set<String>> docsToCheckAndEvents = new HashMap<String, Set<String>>();
+
+        Map<String, Boolean> documentModifiedIgnored = new HashMap<String, Boolean>();
+        for (Event event : events) {
+            log.trace("Proceeding event " + event.getName());
+            EventContext evtCtx = event.getContext();
+            DocumentEventContext docEventCtx = (DocumentEventContext) evtCtx;
+            DocumentModel doc = docEventCtx.getSourceDocument();
+            String docId = doc.getId();
+            if (docEventCtx.getProperties().containsKey(RetentionConstants.RETENTION_CHECKER_LISTENER_IGNORE)
+                    && !documentModifiedIgnored.containsKey(docId)) {
+                // ignore only once per document per bundle, the rule can be attached and document later modified
+                // into
+                // the same transaction
+                documentModifiedIgnored.put(docId, true);
+                continue;
+            }
+            if (!doc.hasFacet(RetentionConstants.RECORD_FACET)) {
+                log.trace("Document is not a record", doc::getPathAsString);
+                continue;
```

Comment:
`log.trace("Document is not a record", doc::getPathAsString);` -> `log.trace("Document {} is not a record", doc::getPathAsString);` --> missing `{}`